### PR TITLE
Fix native testing for xcode 10

### PR DESCRIPTION
**Pull request by ghale**
This fixes the native tests so that they will pass on Xcode 10 and ignore the tests that require 32 bit or both 32 and 64 bit toolchains.

It also allows Xcode 9 toolchains to be discovered if they are added to the system path.

Note that this does NOT fix the issue with extra "swiftdeps~moduleonly" files on Xcode 10.

### CODE REVIEW

File: subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativePreCompiledHeaderIntegrationTest.groovy [hunk 1]
```diff
@@ -540,7 +540,7 @@ abstract class AbstractNativePreCompiledHeaderIntegrationTest extends AbstractIn
     }
 
     String getUniquePragmaOutput(String message) {
-        if (toolChain.displayName == "clang") {
+        if (toolChain.displayName.startsWith("clang")) {
```
Comment (lacasseio) [inline subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativePreCompiledHeaderIntegrationTest.groovy]: ❓ Why was this changed?

File: subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy [hunk 2]
```diff
@@ -320,6 +321,7 @@ abstract class AbstractNativeLanguageIncrementalBuildIntegrationTest extends Abs
     }
 
     @Requires(TestPrecondition.CAN_INSTALL_EXECUTABLE)
+    @RequiresInstalledToolChain(SUPPORTS_32_AND_64)
```
Comment (lacasseio) [inline subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy]: ❓ What is the default requirements? Does it assumes host architecture?

File: subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativePreCompiledHeaderIntegrationTest.groovy (hunk 1 above)
Comment (ghale) [reply subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativePreCompiledHeaderIntegrationTest.groovy]: Because the displayName now shows up as "clang 9.1.0" or "clang 10.0.0" instead of just "clang".

File: subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy (hunk 2 above)
Comment (ghale) [reply subprojects/language-native/src/integTest/groovy/org/gradle/language/AbstractNativeLanguageIncrementalBuildIntegrationTest.groovy]: The assumption is that a toolchain supports a single architecture, which is usually the default architecture of the host (64-bit), but for windows gcc we currently do some magic to make it see the default architecture as 32-bit.  The new requirements should be used wherever a test requires both 32 bit and 64 bit, or wherever a test explicitly declares a 32-bit architecture.
